aws-redshift-cluster/src/main/java/software/amazon/redshift/cluster/Translator.java

```diff
@@ -690,6 +693,44 @@ static String finalClusterSnapshotIdentifierBuilder(String clusterIdentifier, bo
     return null;
   }
 
+  // if we don't provide any track name when we call create-cluster API,
+  // the default track name is "current"
+  // when cx updates the CFN template to include a track name, "current",
+  // we used to call modify-cluster with track("current"),
+  // and we'd get exception "track is already on 'current'".
+  // The following check is to avoid this unnecessary update call
+  static boolean shouldModifyMaintenanceTrack(final ResourceModel oldModel, final ResourceModel newModel) {
+    final String oldTrackName = oldModel.getMaintenanceTrackName();
```

Comment:
oldTrackName is never null. Its either 'current' or whatever the user last provided trackName

Reply:
the model can stay null (the API response is not), that's why we had the issue in the first place

Reply:
https://code.amazon.com/packages/AWSCloudFormationResourceProvidersRedshift/blobs/55a93e83aa0674357e4707ea3d44e7e17b1b5cd9/--/aws-redshift-cluster/src/main/java/software/amazon/redshift/cluster/Translator.java#L646